text/0000-multi-part-examples-in-rustdoc/index.md

```diff
@@ -0,0 +1,231 @@
+- There's no splitting of code blocks.
+
+We imagine a more convenient authoring experience with the suggested design.
+Even though it feels like a close call, that tips the scale.
+
+Also, the suggested approach is not mutually exclusive with this approach, so that this approach could remain a future possibility.
+
+## Tree variant
+
+This design is similar to the suggested design, but blocks may form tree structures. Multiple blocks may continue from a same prior block.
+
+The advantage is that it would allow code sharing between multiple examples.
+
+The disadvantage is that the result might not be reader-friendly.
+Documentation is pages that are consumed usually top to bottom.
+Adding code blocks that continue one another in some tree structure may seem like a good idea from the perspective of the developer/author but would probably not actually be nice for the reader.
+
+## Interleaved variant
+
+This is a feature that is additional to the suggested design, where the blocks of a multi-part example may interleave with blocks of other multi-part examples.
+
+We feel that this could lead to confusing documentation.
+
+## Q: What is the impact of not doing this?
+
+If this feature is not implemented and no other solution is implemented for including text in long code examples, then long code examples in rust documentation continue to be inconvenient for both authors and readers.
+
+# Prior art
+[prior-art]: #prior-art
+
+This RFC is the evolution of a comment analyzing the existing [RFC 2894][rfc_2894]. It follows [this comment][rfc_comment].
+
+If we look at how programming books are commonly formatted, we will often find multi-part code examples.
+They are split into multiple code blocks and continue each other linearly.
+Between the code blocks is usually verbose educational text.
+Rust documentation authors already seem to follow this style while using one or more of the workarounds described earlier.
+This RFC officialy adopts this style.
+
+One could claim that this RFC allows code examples to be written in [literate programming][literate_programming] style.
+
+We are not familiar with a feature equivalent to the doc-tests in other languages.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+We have some todos in the spec section.
+Otherwise, none that we could think of.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+One future possibility is that code editors would support this.
+Assuming, of course, that they would support language features inside of code examples in documentation, at all.
+
+We expect early adoption of this feature.
+
+We expect the release of this feature into stable to improve the overall experience of documenting non-trivial crates.
+
+[current-documented-workaround]: https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-Portions-of-the-example
```

Comment:
This link is broken, it should be `https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example`.

Reply:
Thank you.